### ISSUE TITLE
TextField does not update the `this.state.value` on change.

### ISSUE THREAD

**Issue by hbowden** [DigitalRiver/react-atlas#216]
When the value of TextField is updated, the internal state is not updated. This means that a component using TextField can not get the value of the TextField. In the example below value will always be `"cat"` or whatever is set initially, even if a new value is typed into the TextField.

```Javascript
import React from 'react';
import ReactDOM from 'react-dom';
import { Button, TextField } from 'react-atlas';

export default class Home extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {"number": "cat"};
  }

  clickHandler() {
    console.log("Click: ", this.state.number);
  }
                                                                           
  render() {
    return (
      <div>
        <TextField small 
                   placeholder="Phone Number" 
                   mask="(111) 111-1111" 
                   type={"text"}
                   value={this.state.number}>
        </TextField>
        <Button primary outline onClick={this.clickHandler.bind(this)}>Call</Button>
      </div>
    )
  }
}
```

**Comment by pbalzano91** [comment]
https://github.com/DigitalRiver/react-atlas/pull/218

**Comment by hbowden** [comment]
Fixed in #218
